rules/downgrade-php80/src/Rector/ClassMethod/DowngradeTrailingCommasInParamUseRector.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp80\Rector\ClassMethod;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Function_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\DowngradePhp80\Tests\Rector\ClassMethod\DowngradeTrailingCommasInParamUseRector\DowngradeTrailingCommasInParamUseRectorTest
+ */
+final class DowngradeTrailingCommasInParamUseRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Remove trailing commas in param or use list', [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function __construct(string $value1, string $value2,)
+    {
+        function (string $value1, string $value2,) {
+        };
+
+        function () use ($value1, $value2,) {
+        };
+    }
+}
+
+function inFunction(string $value1, string $value2,)
+{
+}
+CODE_SAMPLE
+                    , <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function __construct(string $value1, string $value2)
+    {
+        function (string $value1, string $value2) {
+        };
+
+        function () use ($value1, $value2) {
+        };
+    }
+}
+
+function inFunction(string $value1, string $value2)
+{
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class, Function_::class, Closure::class];
+    }
+
+    /**
+     * @param ClassMethod|Function_|Closure $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof Closure) {
```

Comment:
We could skip the negation here and run `$this->processUses` only when the instance is `Closure`. Thanks to that we will only have one `return` statement and a simpler condition (without negation).

Reply:
I'm not sure what exactly you mean, the code in words is very hard to read.
Could you maybe diff a suggestion?

Like:

```suggestion
-        if (! $node instanceof Closure) {
+        if (! $node instanceof Closure....) {
```

Reply:
Done in a separate discussion.